src/support.cpp

```diff
@@ -48,6 +48,302 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+            }
+
+            assert(support_infill_part.infill_areas_per_combine_per_density.size() == 0);
+            if (wall_line_count > 0)
+            {
+                // generate insets and use the first inset as the infill area
+                AreaSupport::generateOutlineInsets(
+                    support_infill_part.insets,
+                    support_infill_part.outline,
+                    wall_line_count,
+                    support_line_width);
+                if (support_infill_part.insets.empty())
+                {
+                    continue;
+                }
+                support_infill_part.inner_infill_areas = support_infill_part.insets[0];
+            }
+            else
+            {
+                // for zig zag, we don't want a wall, so we use the outline as the inner infill areas
+                support_infill_part.inner_infill_areas = support_infill_part.outline;
+            }
+
+            // also generate the global inner_infill_area
+            global_inner_support_areas.add(support_infill_part.inner_infill_areas);
+        }
+        global_inner_support_areas = global_inner_support_areas.unionPolygons();
+    }
+
+    // compute different density areas for each support island
+    for (unsigned int layer_nr = 0; layer_nr < total_layer_count - 1; ++layer_nr)
+    {
+        if (layer_nr < min_layer || layer_nr > max_layer)
+        {
+            continue;
+        }
+
+        // generate separate support islands and calculate density areas for each island
+        std::vector<SupportInfillPart>& support_infill_parts = storage.support.supportLayers[layer_nr].support_infill_parts;
+        for (unsigned int part_idx = 0; part_idx < support_infill_parts.size(); ++part_idx)
+        {
+            SupportInfillPart& support_infill_part = support_infill_parts[part_idx];
+
+            // calculate density areas for this island
+            Polygons less_dense_support = support_infill_part.inner_infill_areas; // one step less dense with each density_step
+            for (unsigned int density_step = 0; density_step < max_density_steps; ++density_step)
+            {
+                size_t min_layer = layer_nr + density_step * gradual_support_step_layer_count + layer_skip_count;
+                size_t max_layer = layer_nr + (density_step + 1) * gradual_support_step_layer_count;
+
+                for (float upper_layer_idx = min_layer; static_cast<unsigned int>(upper_layer_idx) <= max_layer; upper_layer_idx += layer_skip_count)
+                {
+                    if (static_cast<unsigned int>(upper_layer_idx) >= total_layer_count)
+                    {
+                        less_dense_support.clear();
+                        break;
+                    }
+                    const Polygons& relevent_upper_polygons = storage.support.supportLayers[upper_layer_idx].inner_support_areas;
```

Comment:
Also:
```
legend:
^ support roof
| support wall
# dense support
+ less dense support

comparing infill    comparing with outline
   ^^^^^^           ^^^^^^
   ####||^^         ####||^^
   ######||^^       ######||^^
   ++++####||       ++++++##||
   ++++++####       ++++++++##
```

I'm not a 100% sure, but I think the second method is better.
Whichever you use, please document clearly why you chose that way of implementing it.
Please explain your choice to me and clearly document the code with your choice.